include/pipettes/core/i2c_writer.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+#include <type_traits>
+#include <variant>
+
+#include "common/core/bit_utils.hpp"
+#include "common/core/buffer_type.hpp"
+#include "common/core/message_queue.hpp"
+#include "pipettes/core/messages.hpp"
+
+namespace i2c_writer {
+
+using namespace pipette_messages;
+
+using TaskMessage = std::variant<std::monostate, WriteToI2C, ReadFromI2C>;
+
+template <template <class> class QueueImpl>
+requires MessageQueue<QueueImpl<TaskMessage>, TaskMessage>
+class I2CWriter {
+    using QueueType = QueueImpl<TaskMessage>;
+
+  public:
+    I2CWriter() = default;
+
+    template <GenericByteBuffer Buffer, typename Data>
+    requires std::is_integral_v<Data>
+    void write(Buffer& buffer, Data data, uint16_t device_address) {
+        add_to_buffer(buffer, data);
```

Comment:
I am a bit concerned with this messaging and memory ownership. `WriteToI2C` holds a pointer to the buffer rather than a copy of the data.  The caller must not deallocate the buffer. 

I think it's a dangerous pattern that could result in mystery crashes.